woocommerce/woocommerce
# [Experimental] Delayed Account Creation Block

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This is the delayed account creation block which will be included on the order confirmation page. This PR includes the new block and baseline functionality behind an experimental build flag.

Closes #50632



### How to test the changes in this Pull Request:

**This block is behind an experimental flag. If you are testing on a production version of WooCommerce this block should not be available (yet).** However you can test:

1. That validation errors during checkout now show a red "i" icon to the left of the text
2. That you can still place a guest email, copy paste the confirmation page URL into a different browser, then input your email address to view the order
3. That the order confirmation page displays correctly without delayed account creation visible 

These instructions are for testing the development build:

1. Appearance > Templates > WooCommerce > Order Confirmation
4. Insert the **Account Creation** block below the order summary (this is were it will be auto inserted in the future). Save.
5. Confirm the editor shows a non-interactive form and an editable list. 
6. You can change the contents of the list if you wish. 
7. Test changing background and text styles, and dark controls on and off.
8. Save changes.
9. On the store frontend, while logged out, add some items to cart and checkout as a guest. Use a new email address.
10. On the confirmation screen check that the account creation block is shown. Enter a password and submit. Ensure that:
    1. Account is created and you are now logged in
    2. Messaging links to the account area
    3. Going to the account area and viewing orders shows the order you just created
    4. That you received the new account email from WooCommerce
    5. Log-out and log-in successfully using the password/email combo you just signed up with.
11. Log-out and go through the checkout again. Checkout as a guest, but use the previous email.
12. The order confirmation should not show the account creation box because the email was already used.
13. Smoke test/try to break the create account form as best you can.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
